### ISSUE TITLE
Confusing error message "expecting string" for syntax error

### ISSUE THREAD

**Issue by ypiguet-epfl** [aseba-community/aseba#742]
In Aseba Studio, many syntax errors are reported as "Expecting string, found xxx instead" where xxx is whatever the parser has found, e.g. "< (smaller than)". "string" is confusing, because it usually refers to a data type or a quoted literal value (unknown in Aseba), which is not the case here. It should probably be "keyword or variable name".
Example:
`var a = 2 < 3
`
In this case, a specific error message would be even better.

**Comment by stephanemagnenat** [comment]
I agree, everything that improves the usability of the compiler is welcome.

However it is not totally trivial, because a string could be a constant name, a variable name, an event name, a function name, a subroutine name. Maybe the term "identifier" would be better, but it should be introduced in the documentation.

Of course, specific error messages are always better.
